### PR TITLE
Add example of bull/bear call spread using index option

### PR DESCRIPTION
**Pull request by LouisSzeto**
#### Description
To add example algorithm of SPXW weekly index options for Bull/Bear Call Spread option strategy.

#### Related Issue
NA

#### Motivation and Context
Lack of combo order examples with index weeklies.

#### Requires Documentation Change
Add the link of example in the Bull/Bear Call Spread option strategy page.

#### How Has This Been Tested?
Bull call spread
- C#: https://www.quantconnect.com/terminal/processCache?request=embedded_backtest_97b791e4fe7212aa248fced841c8ab24.html
- Py: https://www.quantconnect.com/terminal/processCache?request=embedded_backtest_43f213fef12f1c3f6df2428b98cccde4.html

Bear call spread:
- C#: https://www.quantconnect.com/terminal/processCache?request=embedded_backtest_e4069731a06c426802de85d1ec1dae62.html
- Py: https://www.quantconnect.com/terminal/processCache?request=embedded_backtest_8885bb7daa3dd06e9ee4ca58093fc689.html

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Refactor (non-breaking change which improves implementation)
- [ ] Performance (non-breaking change which improves performance. Please add associated performance test and results)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Non-functional change (xml comments/documentation/etc)

#### Checklist:

- [X] My code follows the code style of this project.
- [X] I have read the **CONTRIBUTING** [document](https://github.com/QuantConnect/Lean/blob/master/CONTRIBUTING.md).
- [ ] I have added tests to cover my changes. 
   non-breaking change, `/Data` does not have data for regression
- [X] All new and existing tests passed.
- [X] My branch follows the naming convention `bug-<issue#>-<description>` or `feature-<issue#>-<description>`

### CODE REVIEW

**Review by jhonabreul — APPROVED** [review]
Nice!

**Review by AlexCatarino — APPROVED** [review]

File: Algorithm.CSharp/IndexOptionBullCallSpreadAlgorithm.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        {
+            SetStartDate(2020, 1, 1);
+            SetEndDate(2021, 1, 1);
+            SetCash(100000);
+
+            AddEquity("SPY", Resolution.Minute);
+
+            var index = AddIndex("SPX", Resolution.Minute).Symbol;
+            var option = AddIndexOption(index, "SPXW", Resolution.Minute);
+            option.SetFilter((x) => x.WeeklysOnly().Strikes(-5, 5).Expiration(40, 60));
```
Comment (AlexCatarino) [inline Algorithm.CSharp/IndexOptionBullCallSpreadAlgorithm.cs:37]: Do we need `WeeklysOnly()` for `SPXW`? If not, let's remove it.

File: Algorithm.CSharp/IndexOptionBullCallSpreadAlgorithm.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+            // Get the OptionChain
+            var chain = slice.OptionChains.get(_symbol);
+            if (chain == null) return;
+
+            // Get the nearest expiry date of the contracts
+            var expiry = chain.Min(x => x.Expiry);
+            
+            // Select the call Option contracts with the nearest expiry and sort by strike price
+            var calls = chain.Where(x => x.Expiry == expiry && x.Right == OptionRight.Call).OrderBy(x => x.Strike);
```
Comment (AlexCatarino) [inline Algorithm.CSharp/IndexOptionBullCallSpreadAlgorithm.cs:59]: Please convert it to `ToArray`, and use indices in the creation of the Legs. Below, replace `Count()` with `Length`.

**Review by AlexCatarino — CHANGES_REQUESTED** [review]
Thank you, @LouisSzeto !
Please check out my comment on #6907, and let's try to be consistent across all algorithms.

File: Algorithm.CSharp/IndexOptionBullCallSpreadAlgorithm.cs (hunk 1 above)
Comment (LouisSzeto) [reply Algorithm.CSharp/IndexOptionBullCallSpreadAlgorithm.cs:37]: We need them, otherwise it won't get the weeklies :(

**Review by AlexCatarino — APPROVED** [review]

**Pull request merged**